### PR TITLE
Canary: Add underlines to links in post content and comment content

### PR DESCRIPTION
**Pull request by carolinan**
Adds underlines to links in post content and comment content blocks. With this PR:

- Default state: solid underline
- Hover: no underline
- Active: no underline
- Focus: dashed underline

Partial for [218.](https://github.com/WordPress/twentytwentythree/issues/218)

### CODE REVIEW

File: styles/canary.json [hunk 1]
```diff
@@ -70,6 +70,20 @@
 	},
 	"styles": {
 		"blocks": {
+			"core/comment-content": {
```
Comment (mikachan) [inline styles/canary.json]: This doesn't seem to work for me, these styles are being overridden (by the default link styles I think) and I'm seeing no underline by default. 

image

Swapping the block to `core/comments` seems to fix this.

```suggestion
			"core/comments": {
```

Comment (carolinan) [reply styles/canary.json]: It is intended for the comment content -Links in text that the comment author adds. That is the area that is covered by the accessibility-ready requirement. But that doesn't seem to be working either.

Comment (carolinan) [reply styles/canary.json]: Adding it to core/comments adds it to all links:
Screen Shot 2022-10-06 at 13 56 09

Comment (mikachan) [reply styles/canary.json]: Yeah they didn't seem to work for me either, but they were included when I changed the block to `core/comments`.

Comment (carolinan) [reply styles/canary.json]: Submitted the change.

**Review by mikachan — APPROVED** [review]
Looks good to me!

**Pull request merged**